test/as_documentation/metaconstants.clj

```diff
@@ -145,22 +145,35 @@
     (provided
       --mc-- =contains=> {:c 50000})))
 
+(unfinished gen-doc)
+(fact "Test merging of metaconstant that appear in data and function fakes"
+  (:header (gen-doc)) => "gamma"
+  (provided
+    (gen-doc) => ..doc..
+    ..doc.. =contains=> {:header "gamma"}))
 
+(fact "Test merging of nested metaconstant that appear in data and function fakes"
+  (:header (:raw (gen-doc))) => "gamma"
+  (provided
+    (gen-doc) => {:raw ..doc..}
+    ..doc.. =contains=> {:header "gamma"}))
```

Comment:
Aah, nice example! It breaks the code, though I had to tweak it to be valid midje DSL code because `provided` only applies to the `fact` immediately above it:
```
(fact 
  (vector (:header (gen-doc)) (:header (gen-doc))) => [1 2] 
  (provided 
    (gen-doc) =streams=> [..doc1.. ..doc2..] 
    ..doc1.. =contains=> {:header 1} 
    ..doc2.. =contains=> {:header 2}))
```
results in
```
FAIL at (form-init5605701913344457958.clj:1)
    Expected: [1 2]
      Actual: (nil nil)
       Diffs: in [0] expected 1, was nil
              in [1] expected 2, was nil
```
I'll add this as a `future-fact` test for future implementation